pkg/synthetictests/disruption.go

```diff
@@ -310,3 +309,48 @@ func isOneSecondEvent(eventInterval monitorapi.EventInterval) bool {
 		return true
 	}
 }
+
+func testDNSOverlapDisruption(events monitorapi.Intervals) []*junitapi.JUnitTestCase {
+	const testName = "[sig-network] Disruption should not overlap with DNS problems in cluster running tests"
+	failures := []string{}
+	dnsIntervals := []monitorapi.EventInterval{}
+	disruptionIntervals := []monitorapi.EventInterval{}
+	for _, event := range events {
+		// DNS outage
+		if reason := monitorapi.ReasonFrom(event.Message); strings.HasPrefix(reason, "DisruptionSamplerOutageBegan") {
+			dnsIntervals = append(dnsIntervals, event)
+		}
+		// real disruption
+		if reason := monitorapi.ReasonFrom(event.Message); strings.HasPrefix(reason, "DisruptionBegan") {
+			disruptionIntervals = append(disruptionIntervals, event)
+		}
+	}
+	for _, r := range disruptionIntervals {
+		for _, d := range dnsIntervals {
+			if (r.From.Before(d.To) && d.From.Before(r.To)) || (r.To.Add(10*time.Second).After(d.From) && d.To.Add(10*time.Second).After(r.From)) {
+				failures = append(failures, "Overlap or interval within 10 seconds found.")
```

Comment:
We are appending the same message over and over here.   In the patch I provided earlier that pulls in real intervals data we see 6,107 messages duplicated in the test output.  Let's change this to a count and output the message once with the number of overlaps we detected appended to it.